src/Sulu/Bundle/MediaBundle/Resources/js/views/MediaOverview/MediaOverview.js

```diff
@@ -2,104 +2,149 @@
+            this.setCollectionId(collectionId);
+            this.createCollectionStore(collectionId, this.locale);
+            this.createMediaDatagridStore(collectionId, this.mediaPage, this.locale);
+            this.createCollectionDatagridStore(collectionId, this.collectionPage, this.locale);
+        }
     };
 
-    loadCollectionInfo(collectionId) {
-        return ResourceRequester.get(COLLECTIONS_RESOURCE_KEY, collectionId, {
-            depth: 1,
-            locale: this.locale,
-        }).then(action((collectionInfo) => {
-            const parentCollection = collectionInfo._embedded.parent;
-            this.title = collectionInfo.title;
-            this.parentId = (parentCollection) ? parentCollection.id : undefined;
-        }));
+    @action setCollectionId(id) {
+        this.collectionId = id;
     }
 
-    getTitle() {
-        if (!this.collectionId) {
-            return translate('sulu_admin.all_media');
+    @action createCollectionDatagridStore(collectionId, page, locale) {
+        if (this.collectionDatagridStore) {
+            this.collectionDatagridStore.destroy();
         }
 
-        return this.title;
+        this.collectionDatagridStore = new DatagridStore(
+            COLLECTIONS_RESOURCE_KEY,
+            {
+                page,
+                locale,
+            },
+            (collectionId) ? {parent: collectionId} : undefined
+        );
     }
 
-    @action createCollectionStore(collectionId) {
+    createCollectionStore = (collectionId, locale) => {
         if (this.collectionStore) {
             this.collectionStore.destroy();
         }
 
-        this.collectionId = collectionId;
-        this.collectionStore = new DatagridStore(
-            COLLECTIONS_RESOURCE_KEY,
+        this.collectionStore = new CollectionStore(collectionId, locale);
+    };
+
+    @action createMediaDatagridStore(collectionId, page, locale) {
+        const options = {};
+        options.fields = [
+            'id',
+            'type',
+            'name',
+            'size',
+            'title',
+            'mimeType',
+            'subVersion',
+            'thumbnails',
+        ].join(',');
+
+        page.set(1);
+
+        if (collectionId) {
+            options.collection = collectionId;
+        }
+
+        if (this.mediaDatagridStore) {
+            this.mediaDatagridStore.destroy();
+        }
+
+        this.mediaDatagridStore = new DatagridStore(
+            MEDIA_RESOURCE_KEY,
             {
-                page: this.page,
-                locale: this.locale,
+                page,
+                locale,
             },
-            (collectionId) ? {parent: collectionId} : undefined
+            options,
+            true
         );
     }
 
-    handleOpenFolder = (collectionId) => {
+    handleCollectionOpen = (collectionId) => {
         const {router} = this.props;
         router.navigate(COLLECTION_ROUTE, {id: collectionId, page: '1', locale: this.locale.get()});
```

Comment:
You have renamed the `page` here, haven't you? That's probably the reason why after a back click there is always `page=1` in the URL.